src/gadgets/nonnative/bn/big_uint_mul_mod_chip/mod.rs

```diff
@@ -84,11 +84,11 @@ impl<F: ff::PrimeField> BigUintMulModChip<F> {
     /// - `s_i = lhs_i` where
     ///     - i in [l..k)
     ///     - assumed that `lhs` has more limbs
-    fn assign_sum(
+    pub fn assign_sum<'a>(
         &self,
-        ctx: &mut RegionCtx<'_, F>,
+        ctx: &mut RegionCtx<'a, F>,
         lhs: &OverflowingBigUint<F>,
-        rhs: &[F],
+        rhs: &[impl AssignAdviceFrom<'a, F> + Clone + fmt::Debug],
```

Comment:
For circuit consistency, we always complete with zeros up to the limb limit, because of that we don't needed here to have same size of limbs

Reply:
oh, I don't mean why their sizes are different, but why their types are different and the intent of these types. Probably with an example will be clear.
```
lhs: &OverflowingBigUint<F>,
rhs: &[impl AssignAdviceFrom<'a, F> + Clone + fmt::Debug]
```

Reply:
Just because use case. rhs' can be part of witness & can be cells. But it's not a case if it overflows. At the same time, `lhs` has always overflowed.

We can generalise the function, but to do that we need to refactor `OverflowingBigUint` to store inside `impl AssignAdviceFrom`, not just cells.

Reply:
We can discuss in more detail. It's true that the design is not generalised there, but I would highlight it in an issue so as not to waste time now. 

All usecases are satisfied with the current design, and to reuse it externally we still have to write and write code (examples, documentation, etc.)